### PR TITLE
feat: port check-schema and merge-schema from 1.x

### LINKED ISSUE #23894: add type conflict checker to influx_inspect [port to master]
Port this PR to master https://github.com/influxdata/influxdb/pull/23616

### PR DESCRIPTION
**Pull request by jeffreyssmith2nd**
- Closes #23894

### Description
Adds commands for checking if there any type conflicts in a shard and merging the resulting schema.

### Note for reviewers:
Check the semantic commit type:
 - Feat: a feature with user-visible changes
 - Fix: a bug fix that we might tell a user “upgrade to get this fix for your issue”
 - Chore: version bumps, internal doc (e.g. README) changes, code comment updates, code formatting fixes… must not be user facing (except dependency version changes)
 - Build: build script changes, CI config changes, build tool updates
 - Refactor: non-user-visible refactoring
 - Check the PR title: we should be able to put this as a one-liner in the release notes

### CODE REVIEW

File: cmd/influxd/inspect/type_conflicts/check_schema.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+		root = tc.Path
+	} else {
+		root = path.Dir(tc.Path)
+	}
+	fileSystem := os.DirFS(".")
+	err = fs.WalkDir(fileSystem, root, func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return fmt.Errorf("error walking file: %w", err)
+		}
+		if filepath.Base(path) != "fields.idx" {
```
Comment (davidby-influx) [inline cmd/influxd/inspect/type_conflicts/check_schema.go:117]: Someday we should make the string `fields.idx` a global constant in the tsdb package. Today is probably not that day.

File: cmd/influxd/inspect/type_conflicts/check_schema.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+type TypeConflictChecker struct {
+	Path          string
+	SchemaFile    string
+	ConflictsFile string
+	Logger        *zap.Logger
+
+	logLevel zapcore.Level
+}
+
+func NewCheckSchemaCommand(v *viper.Viper) (*cobra.Command, error) {
```
Comment (davidby-influx) [inline cmd/influxd/inspect/type_conflicts/check_schema.go:30]: We should probably check for the existence of any `tsdb.FieldsChangeFile` and warn or abort.  This change should then be backported to master-1.x (and 1.11 if that branch has been created).  My error in not updating this code after the incremental update to fields.idx (https://github.com/influxdata/influxdb/pull/23728 and https://github.com/influxdata/influxdb/pull/23701).

**Review by davidby-influx — CHANGES_REQUESTED** [review]
Please fix my omissions around `fields.idxl` files.  On reflection, a warning is probably the best strategy, not a halt to processing.

File: cmd/influxd/inspect/type_conflicts/check_schema.go (hunk 2 above)
Comment (jeffreyssmith2nd) [reply cmd/influxd/inspect/type_conflicts/check_schema.go:30]: Added a new commit, I tested it by just touching a `fields.idxl` file next to a `fields.idx` file before running the command.
Screenshot 2022-11-18 at 13 07 02

**Review by davidby-influx — APPROVED** [review]
LGTM

Someday we should move the status messages like this one to stderr.

**Pull request merged**